source/Calamari.Shared/Integration/Packages/Download/S3PackageDownloader.cs

```diff
@@ -64,6 +64,7 @@ string BuildFileName(string prefix, string version, string extension)
             }
 
             var cacheDirectory = PackageDownloaderUtils.GetPackageRoot(feedId);
+            fileSystem.EnsureDirectoryExists(cacheDirectory);
```

Comment:
This is standard across other package downloaders, without it the folder does not exist and downloading fails.